File: MyHordesOptimizerApi/MyHordesOptimizerApi/Dtos/MyHordesOptimizer/CitizenState/CitizenStateTraceDto.cs
using System.Collections.Generic;

namespace MyHordesOptimizerApi.Dtos.MyHordesOptimizer.CitizenState
{
    public class CitizenStateTraceDto
    {
        public CitizenStateDto StartingState { get; set; } = null!;
        public List<CitizenStateStepResultDto> Steps { get; set; } = new();
    }
}

File: MyHordesOptimizerWebsite/src/app/_abstract_model/dto/citizen-state-trace.dto.ts
import { CitizenStateDTO } from './citizen-state.dto';
import { CitizenStateStepDTO } from './citizen-state-step.dto';

export interface CitizenStateStepResultDTO {
    description: string;
    stateAfter: CitizenStateDTO;
}

export interface CitizenStateTraceDTO {
    startingState: CitizenStateDTO;
    steps: CitizenStateStepResultDTO[];
}

export interface CitizenDayStateRequestDTO {
    startingState: CitizenStateDTO;
    steps: CitizenStateStepDTO[];
}
